libminifi/test/archive-tests/MergeFileTests.cpp

```diff
@@ -37,88 +37,112 @@
 #include "../TestBase.h"
 #include "../unit/ProvenanceTestHelper.h"
 
-static const char* FLOW_FILE = "/tmp/minifi-mergecontent";
-static const char* EXPECT_MERGE_CONTENT_FIRST = "/tmp/minifi-expect-mergecontent1.txt";
-static const char* EXPECT_MERGE_CONTENT_SECOND = "/tmp/minifi-expect-mergecontent2.txt";
-static const char* HEADER_FILE = "/tmp/minifi-mergecontent.header";
-static const char* FOOTER_FILE = "/tmp/minifi-mergecontent.footer";
-static const char* DEMARCATOR_FILE = "/tmp/minifi-mergecontent.demarcator";
-
-class ReadCallback: public org::apache::nifi::minifi::InputStreamCallback {
+std::string FLOW_FILE;
+std::string EXPECT_MERGE_CONTENT_FIRST;
+std::string EXPECT_MERGE_CONTENT_SECOND;
+std::string HEADER_FILE;
+std::string FOOTER_FILE;
+std::string DEMARCATOR_FILE;
+
+void init_file_paths() {
+  struct Initializer {
+    Initializer() {
+      static TestController global_controller;
+      char format[] = "/tmp/test.XXXXXX";
+      std::string tempDir = global_controller.createTempDirectory(format);
+      FLOW_FILE = utils::file::FileUtils::concat_path(tempDir, "minifi-mergecontent");
+      EXPECT_MERGE_CONTENT_FIRST = utils::file::FileUtils::concat_path(tempDir, "minifi-expect-mergecontent1.txt");
+      EXPECT_MERGE_CONTENT_SECOND = utils::file::FileUtils::concat_path(tempDir, "minifi-expect-mergecontent2.txt");
+      HEADER_FILE = utils::file::FileUtils::concat_path(tempDir, "minifi-mergecontent.header");
+      FOOTER_FILE = utils::file::FileUtils::concat_path(tempDir, "minifi-mergecontent.footer");
+      DEMARCATOR_FILE = utils::file::FileUtils::concat_path(tempDir, "minifi-mergecontent.demarcator");
+    }
+  };
+  static Initializer initializer;
+}
+
+class FixedBuffer : public org::apache::nifi::minifi::InputStreamCallback {
  public:
-  explicit ReadCallback(uint64_t size) :
-      read_size_(0) {
-    buffer_size_ = size;
-    buffer_ = new uint8_t[buffer_size_];
-    archive_buffer_num_ = 0;
+  explicit FixedBuffer(std::size_t capacity) : capacity_(capacity) {
+    buf_ = new uint8_t[capacity_];
```

Comment:
Is there any reason that prevent using unique ptr here?

Reply:
done